examples/seismic/acoustic/wavesolver.py

```diff
@@ -153,7 +155,7 @@ def gradient(self, rec, u, v=None, grad=None, m=None, **kwargs):
 
         :returns: Gradient field and performance summary
         """
-
+        dt = kwargs.pop('dt', self.dt)
```

Comment:
On a side note, I still don't understand why some fields may be **either** passed as method arguments **or** retrieved from the instance attributes.

Reply:
Not sure what you mean there